### ISSUE TITLE
Documentation contradiction: required backup of keyfile

### ISSUE THREAD

**Issue by rolandu** [borgbackup/borg#6204]
## Have you checked borgbackup docs, FAQ, and open Github issues?

Yes

## Is this a BUG / ISSUE report or a QUESTION?

Documentation BUG (I think)

## Affected documentation

(1) https://borgbackup.readthedocs.io/en/stable/quickstart.html#encrypted-repos

> If the repository is encrypted, you will also need the key and the passphrase (which is protecting the key).
> 
> The key can be located:
> in the repository (repokey mode). Easy, this will usually “just work”.
> in the home directory of the user who did the backup (keyfile mode). This may cause a bit more effort: (...)

(2) https://borgbackup.readthedocs.io/en/stable/usage/init.html

> If you want “passphrase-only” security, use one of the repokey modes. The key will be stored inside the repository (in its “config” file). In above mentioned attack scenario, the attacker will have the key (but not the passphrase).
> 
> If you want “passphrase and having-the-key” security, use one of the keyfile modes. The key will be stored in your home directory (in .config/borg/keys). In the attack scenario, the attacker who has just access to your repo won’t have the key (and also not the passphrase).

(3) https://borgbackup.readthedocs.io/en/stable/faq.html#how-can-i-specify-the-encryption-passphrase-programmatically

> Using keyfile-based encryption with a blank passphrase

(4) [How important is the $HOME/.config/borg directory?](https://borgbackup.readthedocs.io/en/stable/faq.html#how-important-is-the-home-config-borg-directory)

> keys subdirectory
> In this directory all your repository keyfiles are stored. You MUST make sure to have an independent backup of these keyfiles, otherwise you cannot access your backups anymore if you lose them. You also MUST keep these files secret; everyone who gains access to your repository and has the corresponding keyfile (and the key passphrase) can extract it.

## Description of problem

- 1 and 2 tells me that there are different encryption modes: repokey will store the key together with the repo, keyfile will not.
- 3 tells me that there is a special case of keyfile encryption where I don't set a passphrase. Only in this special case will the keyfile be unencrypted. It will be unencryped and only locally stored in that case.
- 4 says that the "keys" subdirectory contains keyfiles and it sounds as if they are *always required* for disaster recovery and *always unencrypted*... which contradicts the above in my opinion.
- I *believe* that 4 applies to keyfile encryption (sentences 1 and 2) and the third sentence ("You also MUST keep these files secret...") applies only to keyfile encryption without a passphrase (a special case). Is that correct?
- Does borg in repokey mode store unencrypted keyfile locally?
- Is a separate backup of the keys subdirectory *required* for repokey encryption?

**Comment by ThomasWaldmann** [comment]
- correct
- the passphrase is totally your choice, using an empty one or a trivial one is of course not protecting the key.
- 4 doesn't say they are always unencrypted.
- 4 (storage in local keys/ directory) applies to keyfile keys and not to repokey keys, correct. this paragraph was recently improved, see PR ...
- no
- backups are never required. but it might really suck if you need one and you don't have one. for the repokey case you could need one in case the `repo_directory/config` file (where the repokey is stored) gets corrupted or lost somehow while all the rest of the repo is still intact. could be due to a software or hardware issue. so better always have a key backup.

**Comment by rolandu** [comment]
May I propose a different wording to make it clearer (see pull request)? I never created a pull request on Github before and I am not fully sure about the proper style of the documentation files... also, it might be too long.

**Comment by mowny** [comment]
> the better way to do key backups is to use this: https://borgbackup.readthedocs.io/en/stable/usage/key.html#borg-key-export

Yeah, about that ... I think there should be an example for --qr-html on there.
I just exported the paper key with `borg key export /borg/borg/ --paper`, which worked.
Then I wanted to export the qr key with `borg key export /borg/borg/ --qr-html`, which complains `output file to export key to expected`, but `borg key export /borg/borg/ --qr-html key.html` errors out with `borg: error: unrecognized arguments: key.html`.
Turns out it has to be `borg key export --qr-html /borg/borg/ key.html` to work. This is not really obvious and belongs in the man page.

**Comment by ThomasWaldmann** [comment]
yeah, add that example to the docs.

also, it now works without the key.html and the just outputs the html on stdout.

**Comment by hexagonrecursion** [comment]
> * 4 says that the "keys" subdirectory contains keyfiles and it sounds as if they are always required for disaster recovery and **always unencrypted**... which contradicts the above in my opinion.

Quote from docs (emphasis mine)
> You also MUST keep these files secret; everyone who gains access to your repository and has the corresponding keyfile (**and the key passphrase**) can extract it.

**Comment by mowny** [comment]
.. and if anyone is updating https://borgbackup.readthedocs.io/en/stable/usage/key.html, maybe they could fix the examples for change-passphrase: Repo is given as -v /path/to/repo on the command line, but the responses have not been edited accordingly, e.g. `Key in "/root/.config/borg/keys/mnt_backup" created.` should be `Key in "/root/.config/borg/keys/path_to_repo" created.`

BTW, is it possible to just move the key to and from ~/.config/borg/keys/path_to_repo to /path/to/repo/config? What happens if (different) keys exist in both? What if a keyfile exists when a new repo is created?

**Comment by hexagonrecursion** [comment]
> .. and if anyone is updating https://borgbackup.readthedocs.io/en/stable/usage/key.html, maybe they could fix the examples for change-passphrase: Repo is given as -v /path/to/repo on the command line, but the responses have not been edited accordingly, e.g. `Key in "/root/.config/borg/keys/mnt_backup" created.` should be `Key in "/root/.config/borg/keys/path_to_repo" created.`

Hello, @mowny. You are drifting kinda off-topic in this thread. I would recommend you to [open a new bug](https://github.com/borgbackup/borg/issues/new).

> BTW, is it possible to just move the key to and from ~/.config/borg/keys/path_to_repo to /path/to/repo/config? What happens if (different) keys exist in both?

Keys in ~/.config/borg/keys/path_to_repo are borg keyfile (or keyfile-blake2) keys, keys in /path/to/repo/config are borg repokey (or repokey-blake2) keys. A repository uses either keyfile or repokey, but never both (or "authenticated" or "none"). You decide the encryption mode when creating a repo with `borg init` and can not change the encryption mode of an existing repo. All this is documented in https://borgbackup.readthedocs.io/en/stable/usage/init.html

> What if a keyfile exists when a new repo is created?

I'll try that and see what happens.
